### PR TITLE
Remove body background ASAP and throw warning

### PR DESCRIPTION
**Pull request by alanorozco**

### CODE REVIEW

File: extensions/amp-lightbox/0.1/amp-lightbox.js [hunk 1]
```diff
@@ -309,6 +316,49 @@ class AmpLightbox extends AMP.BaseElement {
...
+ * @param {!Window} win
+ * @param {!HTMLBodyElement} body
+ * @private
+ */
+function setTransparentBody(win, body) {
+  Services.vsyncFor(win).mutate(() => {
+    setImportantStyles(body, {background: 'transparent'});
+  });
+
+  if (getMode().development && !getMode().test) {
```
Comment (lannka) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: as discussed offline, we can probably avoid a permanent warning by checking:
```js
window.getComputedStyle(document.body)['background-color'] === 'rgba09, 0, 0, 0)'

```

**Comment by lannka** [comment]
For better organizing my PR reviews, I added label "working in progress" to this PR. Please remove the tag once it's ready for review again.

File: extensions/amp-lightbox/0.1/amp-lightbox.js [hunk 2]
```diff
@@ -309,6 +316,58 @@ class AmpLightbox extends AMP.BaseElement {
...
+ */
+function setTransparentBody(win, body) {
+  Services.vsyncFor(win).run({
+    measure(state) {
+      state.alreadyTransparent =
+          computedStyle(win, body)['background-color'] == 'rgba(0, 0, 0, 0)';
+    },
+    mutate(state) {
+      if (!state.alreadyTransparent &&
+          getMode().development && !getMode().test) {
```
Comment (lannka) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: we can probably remove the mode check. it should be a user warning on production

File: extensions/amp-lightbox/0.1/amp-lightbox.js [hunk 3]
```diff
@@ -309,6 +316,58 @@ class AmpLightbox extends AMP.BaseElement {
...
+      }
+
+      // set as !important regardless to prevent changes
+      setImportantStyles(body, {background: 'transparent'});
+    },
+  }, {});
 }
 
+
+if (getMode().runtime == 'inabox') {
```
Comment (lannka) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: why is inabox different for this case?

Comment (lannka) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: Discussed offline, let's put a TODO to clean up the usage of this check.

File: extensions/amp-lightbox/0.1/amp-lightbox.js (hunk 2 above)
Comment (lannka) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: In fact , i believe warn() will be removed from prod binary. So we might use error.

Comment (alanorozco) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: Only `warn()` will be compiled out.

See https://github.com/ampproject/amphtml/blob/master/build-system/runner/src/org/ampproject/AmpCommandLineRunner.java#L54

Removing check and leaving only for `test`.

File: extensions/amp-lightbox/0.1/amp-lightbox.js (hunk 3 above)
Comment (alanorozco) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: Done.

File: extensions/amp-lightbox/0.1/amp-lightbox.js (hunk 2 above)
Comment (lannka) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: ok. but why keep the `test` check?

File: extensions/amp-lightbox/0.1/amp-lightbox.js [hunk 4]
```diff
@@ -309,6 +316,59 @@ class AmpLightbox extends AMP.BaseElement {
...
+      setImportantStyles(body, {background: 'transparent'});
+    },
+  }, {});
 }
 
+
+// TODO(alanorozco): refactor this somehow so we don't need to do a direct
+// getMode() check
+if (getMode().runtime == 'inabox') {
+  setTransparentBody(window, /** @type {!HTMLBodyElement} */ (
```
Comment (lannka) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: can you move this into `maybeSetTransparentBody_()` we we treat inabox the same as FIE, which is our end goal.

Comment (alanorozco) [inline extensions/amp-lightbox/0.1/amp-lightbox.js]: I disagree to that being our end goal. I think our end goal is to warn the author as soon as possible, and "as soon as possible" differs in FIE vs. inabox as you can only determine if under FIE after the element has been attached.

**Pull request merged**